File: iOS-SDK-Demo/ALDemoBaseViewController.m
//
//  ALDemoBaseViewController.m
//  iOS-SDK-Demo
//
//  Created by Thomas So on 9/23/15.
//  Copyright © 2015 AppLovin. All rights reserved.
//

#import "ALDemoBaseViewController.h"

@interface ALDemoBaseViewController ()
@end

@implementation ALDemoBaseViewController

#pragma mark - View Lifecycle

- (void)viewDidAppear:(BOOL)animated
{
    [super viewDidAppear: animated];
    [self.navigationController setToolbarHidden: NO animated: YES];
}

- (void)viewWillDisappear:(BOOL)animated
{
    [self.navigationController setToolbarHidden: YES];
    [super viewWillDisappear: animated];
}


#pragma mark - Logging

- (void)log:(NSString *)message
{
    if ( self.adStatusLabel )
    {
        self.adStatusLabel.text = message;
    }
    ALLog(@"[%@] : %@", NSStringFromClass([self class]), message);
}

@end

File: Swift Demo App/iOS-SDK-Demo/ALDemoBaseViewController.swift
//
//  ALDemoBaseViewController.swift
//  iOS-SDK-Demo
//
//  Created by Thomas So on 9/25/15.
//  Copyright © 2015 AppLovin. All rights reserved.
//

import UIKit

class ALDemoBaseViewController: UIViewController
{
    @IBOutlet weak var adStatusLabel: UILabel?
    
    override func viewDidAppear(animated: Bool)
    {
        super.viewDidAppear(animated)
        self.navigationController?.setToolbarHidden(false, animated: true)
    }
    
    override func viewWillDisappear(animated: Bool)
    {
        self.navigationController?.setToolbarHidden(true, animated: false)
        super.viewWillDisappear(animated)
    }
    
    func log(message: String!)
    {
        dispatch_async(dispatch_get_main_queue()) {
            if let label = self.adStatusLabel {
                label.text = message
            }
        };
        
    }
}
